File: semester-5_replit/oop_task-2/Domain.java
class Domain {
  static private Domain firstObj = null;
  private Domain nextObj = null;

  public Domain() {}
  public Domain(String domainName) {
    this.domainName = domainName;
    if (firstObj == null) {
      firstObj = this;
      return;
    }
    
    Domain p = firstObj;
    while (p.nextObj != null) p = p.nextObj;
    p.nextObj = this;
  }
  static public void showList() {
    Domain p = firstObj;
    while (p != null) {
      p.getInfo();
      p = p.nextObj;
    }
  }

  protected void getInfo() {
    System.out.println(domainName);
    System.out.println();
  }

  protected String domainName;
}

File: course-3-semester-5/visual-studio/oop_lab-2/task-1_cpp/Domain.cpp
#include "Domain.h"

Domain* Domain::lastObj() {
  Domain* last = Domain::listStart;
  while (last->nextObj != nullptr) last = last->nextObj;

  return last;
}

Domain::Domain(std::wstring domainName) {
  this->domainName = domainName;
  
  if (Domain::listStart) lastObj()->nextObj = this;
  else Domain::listStart = this;
}

void Domain::showList() {
  Domain* obj = Domain::listStart;
  while (obj != nullptr) {
    obj->getInfo();
    obj = obj->nextObj;
  } 
}

void Domain::getInfo() {
  std::wcout << this->domainName << L"\n\r";
  std::wcout << L"\n\r";
}
